### ISSUE TITLE
exported: report differently a method that case was altered

### ISSUE THREAD

**Issue by ccoVeille** [mgechev/revive#1236]
```go
package sandbox

// Foo is OK
func Foo() {}

// FoobarQuux was renamed to FooBarQuux
func FooBarQuux() {}
```

The classic example of this is a method where an acronym or a [Go initialisms](https://go.dev/wiki/CodeReviewComments#initialisms) was fixed

`CheckJson => CheckJSON`

revive reports the right things yet, but it could be improved.

![Image](https://github.com/user-attachments/assets/c39b81f4-a5cf-48f9-9e01-b0d55eef2a93)

But because golangci-lint exclude [`EXC0012`](https://golangci-lint.run/usage/false-positives/#exc0012) by default.
This one is not reported via golangci-lint.

We should consider trying to detect it, and report another message that won't be caught by `EXC0012`

Here the idea is to compare the lowercase name of the method, we can be 100% sure the method was renamed.

Please note #1235 is a special case of #1236

- #1235
- #1236

**Comment by chavacava** [comment]
IMO there is not enough aggregated value to justify the enhancement (but it's only my opinion)

**Comment by ccoVeille** [comment]
Let's stick with #1235 then, and use a wording that fits both, OK for you?
